### DISCUSSION TITLE
Minimal cluster setup with docker compose

### DISCUSSION THREAD

**Discussion by tomiszili** [valkey-io/valkey#1240] [Q&A]
Hello,

I want to get some help about how should I start with a minimal cluster setup with docker compose. I want to test my application in a containerized way. But I'm a bit lost what CMD and config should I run the Valkey official container to create  1primary+1replica cluster with TLS.
I would really appreciate if you can share some exact code with me.

Thanks in advance!

**Comment by amirreza-sf80** [comment]
[compose](https://github.com/valkey-io/valkey-py/blob/main/compose.yaml)
[docker file](https://github.com/valkey-io/valkey-py/tree/main/dockers)

the valkey py tests could be of help
see if it works

**Comment by tomiszili** [comment]
@amirreza8002 thanks for your quick help.
Is that even possible to create a single primary node with 1 replica within the cluster mode? Or there is a limitation of 6 nodes (3 primary 3 replica) if i want a minimal cluster setup?

**Comment by amirreza-sf80** [comment]
Hi
i don't think you can
at least my local setup errored when i did less than 6
it kinda makes sense, since cluster of 1 is just a normal server

but who knows, you might be able to find a workaround
doing a search on redis, i don't see a workaround tho

**Comment by tomiszili** [comment]
Hello,

How can i create a cluster with TLS via valkey-cli?
